### PR TITLE
Feature/macgills/1301 night mode

### LINKED ISSUE #1028: Night Mode Tab should be gray in night mode
**Describe the bug**
Night mode Tab should be gray in Night Mode because its not clickable.

**Screenshots**
screenshot_20190223-193016

**Environment**
- Version of Kiwix Android : latest master
- Device : Asus m1
- OS version :API 27

### LINKED ISSUE #1301: Night mode does not work properly in library (with 3.0)
**Describe the bug**
Night mode inverts book icon colour in 'Get Content' library

**Steps to reproduce the behavior:**
Navigate to 'Get Content' library with in-app night mode enabled

**Screenshots**
<p align="center">
  
</p>

**Environment**
- Version of Kiwix Android : Tested until [this](https://github.com/kiwix/kiwix-android/commit/9c1d36657a7a476a1625d9f0a49a573a16eba322) commit on `develop`
- Device : Moto G5+
- OS version : 8.1.0

### LINKED ISSUE #1573: Bookmarks not readable in nightmode
**Describe the bug**
If I switch to night mode the bookmarks are white text on white background. Is the background left in day mode and the text is switched to night mode?

**Expected behavior**
The text should be white and the background should be black or grey like the layout of the settings menu in night mode.

**Steps to reproduce the behavior:**
1. make sure you have a bookmark
2. switch to night mode or use auto night mode and slip under the covers.
3. try to read the bookmark list

**Environment**
- Version of Kiwix Android : 
3.0.4 Build: 230004 release and nightly from 2019-10-27 [3423-universal-debug.apk](https://download.kiwix.org/nightly/2019-10-27/3423-universal-debug.apk)
- Device : Samsung Galaxy S5 SM-G900F i9300
- OS version : lineage_klte-userdebug 9 PQ3A. 190801.002 750bd77430

### PR DESCRIPTION
**Pull request by macgills**
Fixes #1573 
Fixes #1301 
Fixes #1028 

Old night mode implementation removed and theme based version implemented, based heavily on [material theme builder](https://github.com/material-components/material-components-android/tree/master/material-theme-builder)

### CODE REVIEW

**Comment by macgills** [comment]
Also by integrating with MaterialComponents we now have a [typography system](https://material.io/design/typography/the-type-system.html#type-scale) and should probably make a ticket to make all our textviews compliant with it

File: app/src/main/java/org/kiwix/kiwixmobile/language/LanguageActivity.kt [hunk 1]
```diff
@@ -69,7 +68,7 @@ class LanguageActivity : BaseActivity() {
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     setContentView(R.layout.activity_language)
-    setSupportActionBar(toolbar)
+    setSupportActionBar(findViewById(R.id.toolbar))
```
[the remark this answers is not in the document]
Comment (macgills) [reply app/src/main/java/org/kiwix/kiwixmobile/language/LanguageActivity.kt]: This is unfortunate. You can't access a view by the synthetic extension functions if the id is not in your module.
It seems like google is going to force [view binding](https://developer.android.com/topic/libraries/view-binding) down our throats and that might resolve this, just looks and feels uglier

**Comment by kelson42** [comment]
@macgills Works fine for me... but the welcome page which has been forgotten :(

**Comment by kelson42** [comment]
There is also a problem with the first picture of Youtube based ZIM file. Here an example with the ZIM "Mali pour les Nuls"

screenshot

**Comment by macgills** [comment]
@kelson42 the welcome page is a bit annoying, we would have to supply new resources for the plane animation and the `kiwix-with-logo`.png.

An alternative is that we just force this screen to be `light` themed for now and create a ticket to add support for this screen and its assets later on.

How Dark Mode works in WebView
```
  public void activateNightMode() {
    Paint paint = new Paint();
    ColorMatrixColorFilter filterInvert = new ColorMatrixColorFilter(NIGHT_MODE_COLORS);
    paint.setColorFilter(filterInvert);
    setLayerType(LAYER_TYPE_HARDWARE, paint);
  }
```
How the images in the webview get inverted
```
 private const val INVERT_IMAGES_VIDEO =
      "img, video { \n -webkit-filter: invert(1); \n filter: invert(1); \n} \n"

 if ("text/css" == mime.value && nightModeConfig.isNightModeActive()) {
                outputStream.write(INVERT_IMAGES_VIDEO.toByteArray(Charsets.UTF_8))
              }
```
I'll need to investigate the specific html that is getting written from the thumbnail and expand this? No expert in web content.

**Comment by kelson42** [comment]
@macgills Creating a new resource should not be a big deal?!

**Comment by macgills** [comment]
It isn't but it is not something an android dev should be responsible for. I am not a graphic designer, I don't have photoshop or sketch, I've got paint and gimp (and I might have deleted gimp).

I also don't know how you want the icons to look in dark mode
image
Do we want to change the icon to white? Invert the colors? Supply a grey-ish frame? Add a white outline? The inversion for the kiwix banner is kinda rough on the eyes. This current `kiwix_icon_with_title` is also unspecified for any density so the app is currently using the same image on all screen densities.

image

The airplane icon is a bit easier because it is a vector drawable, I can supply another resource where I change the colors but what color? The grey outline is `#545d79` and the light blue pilot is `#b5d9f6`.

**Comment by macgills** [comment]
I have been able to get the Video Preview to invert its colors but the header is proving more annoying
image
Really don't know why it isn't taking the inversion

**Comment by macgills** [comment]
@kelson42 found out why it is not taking the inversion, I had the completely wrong element selected

image

the div having an id `header` means a css style adds a `background-image`. This presents a slight problem because I cannot select the div by the attribute `background-image` (or at least `div[background-image]` did not work) so I have to use the id `#header`/`div#header` which I am a bit unsure of

**Comment by kelson42** [comment]
> It isn't but it is not something an android dev should be responsible for. I am not a graphic designer, I don't have photoshop or sketch, I've got paint and gimp (and I might have deleted gimp).

I have Gimp. Please tell me the path in the source code for which you want an inverted version and I will do it.

**Comment by macgills** [comment]
@kelson42 `kiwix-android\core\src\main\res\drawable\kiwix_icon_with_title` but I don't think we want to invert it as it will be all white, if you are happy for the kiwix icon to change color then by all means go for it.

**Pull request merged**